pkg/metrics/experimental_register.go

```diff
@@ -0,0 +1,40 @@
+// +build experimental_metrics
+
+package metrics
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	// Register experimental metrics
+	reconcileMetrics = reconcileCounters(operatorController, adoptionCSVController, adoptionSubscriptionController, operatorConditionController, operatorConditionGeneratorController)
+	registerReconcileMetrics()
+}
+
+func reconcileCounters(reconcilerNames ...string) map[string]prometheus.CounterVec {
+	result := map[string]prometheus.Counter{}
```

Comment:
Should this be prometheus.CounterVec{} now that we've updated them from counters to counter vectors?

Reply:
Ah, yeah looking at the e2e-test output it wasn't able to compile:

```
go build -mod=vendor -ldflags "-X github.com/operator-framework/operator-lifecycle-manager/pkg/version.GitCommit=22d1a703fea54d1b562dfe99f1edc110e06d17f5 -X github.com/operator-framework/operator-lifecycle-manager/pkg/version.OLMVersion=`cat OLM_VERSION`" -tags "json1 experimental_metrics" -o bin/catalog github.com/operator-framework/operator-lifecycle-manager/cmd/catalog
# github.com/operator-framework/operator-lifecycle-manager/pkg/metrics
pkg/metrics/experimental_register.go:21:13: cannot use reconcileCounter(s) (type prometheus.CounterVec) as type prometheus.Counter in assignment:
	prometheus.CounterVec does not implement prometheus.Counter (missing Add method)
pkg/metrics/experimental_register.go:23:2: cannot use result (type map[string]prometheus.Counter) as type map[string]prometheus.CounterVec in return argument
pkg/metrics/experimental_register.go:27:33: cannot use prometheus.NewCounterVec(prometheus.CounterOpts{...}, []string{...}) (type *prometheus.CounterVec) as type prometheus.CounterVec in return argument
make: *** [Makefile:105: github.com/operator-framework/operator-lifecycle-manager/cmd/catalog] Error 2
```

Reply:
My mistake, fixed this.